[RickyHFR] iP

RickyHFR opened the pull request:
# DukePro: Your Ultimate Task Management Tool 🚀

DukePro is a **lightweight**, **text-based** task manager designed to free your mind from the burden of remembering tasks. As David Allen wisely said:

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://gettingthingsdone.com))

With DukePro, you can focus on what truly matters while it handles the rest. Here's why you'll love it:

- **Simple and intuitive**: Easy to learn and use.
- **Blazing fast**: SUPER FAST performance for seamless task management.
- **Free forever**: No hidden costs, no subscriptions.

## Key Features

1. **Task Management**: Add, edit, and organize your tasks effortlessly.
2. **Deadlines (Coming Soon)**: Stay on top of your schedule with deadline tracking.
3. **Reminders (Coming Soon)**: Never miss a task with timely reminders.

## Getting Started

To start using DukePro, follow these steps:

1. Download it from [here](#).
2. Double-click the application to launch it.
3. Add your tasks using the simple text interface.
4. Let DukePro manage your tasks for you. 😉

## Example Usage

Here’s a quick example of how to add a task in DukePro:

```plaintext
1. [x] Buy groceries
2. [ ] Finish the report
3. [ ] Call John

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

RickyHFR committed:
Rename chatbot and add basic greeting

RickyHFR committed:
Echo

RickyHFR committed:
add list

RickyHFR committed:
mark as done

RickyHFR committed:
some bugs for event and ddl

RickyHFR committed:
Level-4 bugs fixed

RickyHFR committed:
A-TextUiTesting and bug fixing

RickyHFR committed:
bug fix

RickyHFR committed:
Handle Errors

RickyHFR committed:
delete function

RickyHFR committed:
small updates

RickyHFR committed:
initialize storage (incomplete)

RickyHFR committed:
save

RickyHFR committed:
Merge branch 'branch-Level-7'
Incorporate changes from branch-Level-7.

RickyHFR committed:
Add dates to Event and Deadline tasks

RickyHFR committed:
Add time and fix bugs

RickyHFR committed:
Merge branch-Level-8 into master

RickyHFR committed:
Make implementation more OOP by seperating classes

RickyHFR committed:
Merge branch-A-MoreOOP

RickyHFR committed:
Make proper packages

RickyHFR committed:
Merge branch-A-Packages

RickyHFR committed:
Merge branch add-gradle-support

RickyHFR committed:
Use Gradle

RickyHFR committed:
Merge branch-A-Gradle into master

RickyHFR committed:
Fix some bugs related to handling exceptions

RickyHFR committed:
merge the master branch for some bugs fixing

RickyHFR committed:
Add testing files

RickyHFR committed:
Merge branch-A-JUnit

RickyHFR committed:
Package the app into JAR

RickyHFR committed:
Merge branch-A-Jar

RickyHFR committed:
Fix small bug related to welcome message

RickyHFR committed:
Merge the master branch

RickyHFR committed:
Add JavaDoc

RickyHFR committed:
Add JavaDoc

RickyHFR committed:
Tweak coding style

RickyHFR committed:
Add find feature

RickyHFR committed:
Merge branch-A-JavaDoc

RickyHFR committed:
Resolved merge conflicts and completed merge branch-A-JavaDoc

RickyHFR committed:
Merge branch 'master' of https://github.com/RickyHFR/ip

RickyHFR committed:
Resolve merge conflicts from branch-Level-9

RickyHFR committed:
Incorporate CheckStyle into project

A new CheckStyle configuration file has been added to the project. The configuration file is located at `config/checkstyle/checkstyle.xml`. The configuration file is used to enforce a consistent coding style across the project. The configuration file is based on the Google Java Style Guide.
The CheckStyle configuration file is used to check the code for compliance with the Google Java Style Guide. The configuration file is used to enforce a consistent coding style across the project. The configuration file is based on the Google Java Style Guide.
Modifications have been made to adhere to the Google Java Style Guide.

RickyHFR committed:
Merge branch 'branch-A-CheckStyle'

Resolved Checkstyle violations across the codebase.

- Fixed indentation issues in Java files.
- Added missing Javadoc comments for public methods.
- Corrected line length violations by breaking long lines.
- Removed unused imports and variables.

These changes ensure compliance with the project's coding standards
and improve code readability.

RickyHFR committed:
Add GUI using JavaFX

A GUI has been added to the application using JavaFX.
The GUI is designed to be user-friendly and intuitive, allowing users to interact with the application more easily.
The GUI includes a main window that displays the user's tasks, a dialog box for user input, and a launcher class that initializes the application.
The GUI is designed to be responsive and interactive, providing users with a seamless experience when using the application.

Currently the GUI is half done, the main window is able to display the tasks and the dialog box is able to take in user input.
However, more decorations and features will be added in the future to enhance the user experience.

RickyHFR committed:
Complete the fundamental structure of adding GUI

Add the basic structure of the GUI, including the main window and the dialog box.
Add css files for styling the GUI components.
Add the necessary methods to handle the GUI components.

RickyHFR committed:
Fix style issues in DialogBox

Future work:
- Improve the style of the DialogBox
- Add name labels to the DialogBox

RickyHFR committed:
Merge branch-Level-10

LeeJiaWei23 commented on src/test/java/ricky/task/DeadlineTest.java:
```diff
@@ -0,0 +1,24 @@
+
+import org.junit.jupiter.api.Test;
+
+public class DeadlineTest {
+    @Test
+    public void testToString() {
+        LocalDateTime by = LocalDateTime.parse("2023-12-31T23:59", DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+        Deadline deadline = new Deadline("submit report", by);
+        assertEquals("[D][ ] submit report (by: Dec 31 2023 11:59PM)", deadline.toString());
+    }
+
+    @Test
+    public void testStore() {
+        LocalDateTime by = LocalDateTime.parse("2023-12-31T23:59", DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+        Deadline deadline = new Deadline("submit report", by);
+        assertEquals("D | 0 | submit report | 2023-12-31T23:59", deadline.storeInfo());
+    }
+}
```
Perhaps you could consider changing test method names to `unitBeingTested_descriptionOfTestInputs_expectedOutcome`.

LeeJiaWei23 reviewed (commented):
LGTM. Code is clean. Just 1 small detail. 👍

waylonggggg commented on src/main/java/ricky/Parser.java:
```diff
@@ -0,0 +1,89 @@
+     * @return The command corresponding to the user input.
+     * @throws RickyException If the user input is invalid.
+     */
+    public static Command parse(String input) throws RickyException {
```
Perhaps the method can be made shorter?

waylonggggg commented on src/main/java/ricky/Storage.java:
```diff
@@ -0,0 +1,124 @@
+     * @throws RickyException If the file contains invalid tasks.
+     * @throws IOException If an I/O error occurs.
+     */
+    public ArrayList<Task> loadTasks() throws RickyException, IOException {
```
This as well, the code could be condensed using helper functions

waylonggggg commented on src/main/java/gui/DialogBox.java:
```diff
@@ -0,0 +1,79 @@
+        }
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
```
A bit of a nit, but perhaps the parameter "img" could be renamed to an english word e.g. "image", as per the given coding standards

waylonggggg reviewed (commented):
LGTM, just a few nits here and there

RickyHFR committed:
Add some assertions to the Parser class

Assertions are added to the Parser class to ensure that the input is valid.

RickyHFR committed:
Add welcome message to Ricky.java

I added a welcome message to Ricky.java at the beginning of the main method.

RickyHFR committed:
Remove uncessary Assertions

Since some error does not need to stop the program,
we can remove the assertions to make the program more robust.

RickyHFR committed:
Fix bugs about the storage of tasks

Remaining bugs include:
- Saving tasks when closing from GUI is not achieved

Will fix in the future

RickyHFR committed:
Add some assertions

Make use of assertions to ensure that the index is within the bounds of the list in various
operations

RickyHFR committed:
Improve code quality by refactoring code

Add JavaDoc comments to all classes and methods

RickyHFR committed:
Merge branch 'master' into branch-A-Assertions

RickyHFR committed:
Merge pull request #1 from RickyHFR/branch-A-Assertions

Branch a assertions

RickyHFR committed:
Merge branch 'master' into branch-A-CodeQuality

RickyHFR committed:
Merge pull request #2 from RickyHFR/branch-A-CodeQuality

Branch a code quality

RickyHFR committed:
Simplify Parser.java and Storage.java

I have simplified the code in Parser.java and Storage.java to make it more readable and maintainable.
I will now proceed to test the changes to ensure that they work as expected.

RickyHFR committed:
Set up Continuous Integration (CI)

- Create a new file in the .github/workflows directory called gradle.yml

RickyHFR committed:
Make the command syntax more flexible

Provide shorter aliases for keywords
t -> todo
d -> delete
e -> event
ddl -> deadline
f -> find
l -> list
m/um -> mark/unmark

RickyHFR committed:
Merge branch 'branch-A-Extension'

RickyHFR committed:
Fix small buts in DeleteCommand

- Fix typo in DeleteCommand

RickyHFR committed:
Update README.md and add Ui.png

RickyHFR committed:
Fix small naming issue in Ricky.java

I have changed the name of the variable from 'firstTime' to
'isFirstTime' in the Ricky.java file.

RickyHFR committed:
Update README.md

I added a summary of the commands in the README.md file.
This allows users to quickly see the commands that are available.

RickyHFR committed:
Fix small bugs on Storage.java

Ensure that the App is able to run without any errors in both
Windows and Mac OS.
